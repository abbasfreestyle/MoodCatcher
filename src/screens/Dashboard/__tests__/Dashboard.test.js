import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DashboardScreen } from '..';

const requiredProps = {
  screenProps: {}
};

const defaultState = {
  loading: true,
  error: false,
  averageMood: 0,
  entries: 0,
  list: []
};

const wrapper = shallow(<DashboardScreen {...requiredProps} />);

describe('Screens > Dashboard', () => {
  describe('Snapshots', () => {
    afterEach(() => {
      wrapper.setState(defaultState);
    });

    test('should match with default state', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when loading is true', () => {
      wrapper.setState({ loading: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when error is true', () => {
      wrapper.setState({ loading: false, error: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when list is empty', () => {
      wrapper.setState({
        loading: false,
        error: false,
        list: [],
        entries: 0,
        averageMood: 0
      });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('should match when list is not empty', () => {
      const mockResponse = [
        {
          comment: 'Test comment',
          date: new Date(2019, 3, 25).valueOf(),
          feelings: {
            items: [{ id: 1, name: 'Relaxed' }, { id: 2, name: 'Happy' }]
          },
          mood: 6
        }
      ];
      wrapper.setState({
        loading: false,
        list: mockResponse,
        error: false,
        entries: mockResponse.length,
        averageMood: 75
      });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
