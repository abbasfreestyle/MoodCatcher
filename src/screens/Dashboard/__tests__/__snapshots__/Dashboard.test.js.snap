// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens > Dashboard Snapshots should match when error is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Row
    alignItems="center"
    justifyContent="space-between"
  >
    <Title>
      Dashboard
    </Title>
    <View
      marginRight={10}
    >
      <Regular
        disabled={false}
        flex={false}
        margin={0}
        onPress={[Function]}
      >
        Sign Out
      </Regular>
    </View>
  </Row>
  <Text>
    Oops! Sorry, something went wrong.
  </Text>
</View>
`;

exports[`Screens > Dashboard Snapshots should match when list is empty 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Row
    alignItems="center"
    justifyContent="space-between"
  >
    <Title>
      Dashboard
    </Title>
    <View
      marginRight={10}
    >
      <Regular
        disabled={false}
        flex={false}
        margin={0}
        onPress={[Function]}
      >
        Sign Out
      </Regular>
    </View>
  </Row>
  <Text>
    No moods recorded :(
  </Text>
</View>
`;

exports[`Screens > Dashboard Snapshots should match when list is not empty 1`] = `
<FlatList
  ListHeaderComponent={
    <View>
      <Row
        alignItems="center"
        justifyContent="space-between"
      >
        <Title>
          Dashboard
        </Title>
        <View
          marginRight={10}
        >
          <Regular
            disabled={false}
            flex={false}
            margin={0}
            onPress={[Function]}
          >
            Sign Out
          </Regular>
        </View>
      </Row>
      <Row
        alignItems="center"
        justifyContent="space-evenly"
      >
        <AnimatedCircularProgress
          arcSweepAngle={270}
          backgroundColor="#D4D4D5"
          duration={500}
          easing={[Function]}
          fill={75}
          lineCap="round"
          prefill={0}
          rotation={-135}
          size={100}
          tintColor="#A073B2"
          width={10}
        >
          [Function]
        </AnimatedCircularProgress>
        <Column>
          <View>
            <Title>
              Score 
              75
              %
            </Title>
          </View>
          <Title>
            mood(s) 
            1
          </Title>
        </Column>
      </Row>
    </View>
  }
  data={
    Array [
      Object {
        "comment": "Test comment",
        "date": 1556146800000,
        "feelings": Object {
          "items": Array [
            Object {
              "id": 1,
              "name": "Relaxed",
            },
            Object {
              "id": 2,
              "name": "Happy",
            },
          ],
        },
        "mood": 6,
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`Screens > Dashboard Snapshots should match when loading is true 1`] = `
<View
  alignItems="center"
  flex={1}
  justifyContent="center"
>
  <Loading
    size={200}
  />
</View>
`;

exports[`Screens > Dashboard Snapshots should match with default state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Row
    alignItems="center"
    justifyContent="space-between"
  >
    <Title>
      Dashboard
    </Title>
    <View
      marginRight={10}
    >
      <Regular
        disabled={false}
        flex={false}
        margin={0}
        onPress={[Function]}
      >
        Sign Out
      </Regular>
    </View>
  </Row>
  <Text>
    Oops! Sorry, something went wrong.
  </Text>
</View>
`;
